include/arch/arm/arch/64/mode/object/structures.h

```diff
@@ -236,60 +181,29 @@ static inline void *CONST cap_get_archCapPtr(cap_t cap)
     }
 }
 
-static inline bool_t pgde_pgde_pud_ptr_get_present(pgde_t *pgd)
-{
-    return (pgde_ptr_get_pgde_type(pgd) == pgde_pgde_pud);
-}
-
-static inline bool_t pude_pude_pd_ptr_get_present(pude_t *pud)
-{
-    return (pude_ptr_get_pude_type(pud) == pude_pude_pd);
-}
-
-static inline bool_t pude_pude_1g_ptr_get_present(pude_t *pud)
+static inline bool_t pte_pte_page_ptr_get_present(pte_t *pt)
 {
-    return (pude_ptr_get_pude_type(pud) == pude_pude_1g);
+    return (pte_ptr_get_pte_type(pt) == pte_pte_page);
 }
 
-static inline pude_t pude_invalid_new(void)
+static inline bool_t pte_pte_table_ptr_get_present(pte_t *pt)
 {
-    return (pude_t) {
-        {
-            0
-        }
-    };
+    return (pte_ptr_get_pte_type(pt) == pte_pte_table);
 }
 
-static inline bool_t pde_pde_small_ptr_get_present(pde_t *pd)
+static inline bool_t pte_4k_page_ptr_get_present(pte_t *pt)
 {
-    return (pde_ptr_get_pde_type(pd) == pde_pde_small);
+    return (pte_ptr_get_pte_type(pt) == pte_pte_4k_page);
 }
 
-static inline bool_t pde_pde_large_ptr_get_present(pde_t *pd)
+static inline bool_t pte_ptr_get_valid(pte_t *pt)
 {
-    return (pde_ptr_get_pde_type(pd) == pde_pde_large);
+    return (pte_ptr_get_pte_type(pt) != pte_pte_invalid);
 }
 
-static inline pde_t pde_invalid_new(void)
+static inline uint64_t pte_page_ptr_get_page_base_address(pte_t *pt)
 {
-    return (pde_t) {
-        {
-            0
-        }
-    };
+    uint64_t ret;
+    ret = (pt->words[0] & 0xfffffffff000ull);
+    return ret;
 }
```

Comment:
I probably knew this at some point, but is there a reason why we're not using the bitfield generator function for this one? `page_base_address` is a proper field, so should work, no?

Also, would we not be catching the `UXN` field with this mask? (`pte_sw_type` is always 0 for `page_pte`, but `UXN`, I'm less sure)

Reply:
Actually I think I miscounted, the mask is fine, should be 36 bits, which fits and does mask out the higher parts (UXN etc).

I'd still go for the bitfield function unless there is a specific reason not to.

Reply:
> I probably knew this at some point, but is there a reason why we're not using the bitfield generator function for this one? `page_base_address` is a proper field, so should work, no?

Because the asserts that check the type bits line up fail if using the generated functions.